File: Python/Punto_6.py
# Se importa el modulo random para generar números pseudoaleatorios
import random
# Se solicita ingresar el tamaño del arreglo
n=int(input("Ingrese un tamaño para el arreglo: "))
# Se crea un arreglo del tamaño solicitado lleno de ceros
lista=[0]*n
"""Se recorre cada posición del arreglo con map, cambiando el elemento
   por un número entre -100 y 100"""
resultado=list(map(lambda elemento:random.randint(-100,100),lista))
# Se imprime la lista generada
print(resultado)
# Se guardan el número mayor y menor de la lista en maximo y minimo
maximo=max(resultado)
minimo=min(resultado)
# Se imprimen los elementos mayor y menor
print(f"Máximo: {maximo}")
print(f"Mínimo: {minimo}")

File: Java/Punto6.java
package Java;

import java.util.List;
import java.util.Random;
import java.util.Scanner;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class Punto6 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingrese longitud del arreglo: ");
        int n = scanner.nextInt();

        // Generar lista con números aleatorios usando el paradigma funcional
        List<Integer> lista = Stream.generate(() -> new Random().nextInt(100)) // Números aleatorios entre 0 y 99
                .limit(n)
                .collect(Collectors.toList());

        // Obtener el máximo y mínimo usando Streams
        int max = lista.stream().max(Integer::compareTo).orElseThrow();
        int min = lista.stream().min(Integer::compareTo).orElseThrow();

        System.out.println("Lista generada: " + lista);
        System.out.println("Número Máximo: " + max);
        System.out.println("Número Mínimo: " + min);
    }
}